addons/account/models/account_move.py

```diff
@@ -2270,6 +2270,13 @@ def action_post(self):
     def js_assign_outstanding_line(self, line_id):
         self.ensure_one()
         lines = self.env['account.move.line'].browse(line_id)
+        if not lines.currency_id and self.currency_id != self.company_id.currency_id and not lines.matched_debit_ids and not lines.matched_credit_ids:
+            amount_currency = self.company_id.currency_id._convert(lines.balance, self.currency_id, self.company_id, lines.date or fields.Date.today())
```

Comment:
`lines.date` is a required field. Then, ` or fields.Date.today()` is useless

Reply:
Removed.